sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -2451,9 +2457,15 @@ pi_result piEventRelease(pi_event Event) {
       // If the fence associated with this command list has signalled, then
       // Reset the Command List Used in this event and put it back on the
       // available list.
-      Event->Queue->ZeCommandListFenceMapMutex.lock();
-      Event->Queue->resetCommandListFenceEntry(Event->ZeCommandList, true);
-      Event->Queue->ZeCommandListFenceMapMutex.unlock();
+      if (Event->Queue->ZeCommandQueue) {
```

Comment:
It looks like this is guarding against the case when queue was already release in piQueueRelease, but how is it possible? @romanovvlad doesn't it look like a problem in SYCL RT that queue is released before the events created for the commands in that queue?

Reply:
We've discussed this offline and currently SYCL RT assumes this is a legal thing to do because it was driven by OpenCL being able to handle this due to internal reference counting, which keeps the queue alive until events in it are done. 

https://www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/clReleaseCommandQueue.html

> After the command_queue reference count becomes zero **and all commands queued to command_queue have finished** (eg. kernel-instances, memory object updates etc.), the command-queue is deleted.

We lack this functionality in PI Level-Zero plugin, not so sure about CUDA plugin. It seems that in this case it is easier to request in PI API that queue is only released after sibling events are released. @nrspruit, if this isn't straight-forward to fix SYCL RT then please add a TODO comment here, because currently we will not add the entry back to the pool.